source/material_model/reactive_fluid_transport.cc

```diff
@@ -86,23 +86,24 @@ namespace aspect
                     std::vector<double> &melt_fractions,
                     const MaterialModel::MaterialModelOutputs<dim> *out) const
     {
-      const std::shared_ptr<const MeltOutputs<dim>> fluid_out = out->template get_additional_output_object<MeltOutputs<dim>>();
+      std::shared_ptr<const MeltOutputs<dim>> fluid_out;
+      if (out != nullptr)
+        fluid_out = out->template get_additional_output_object<MeltOutputs<dim>>();
+
       if (fluid_out != nullptr && in.requests_property(MaterialProperties::additional_outputs))
         {
+          const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
           for (unsigned int q=0; q<in.n_evaluation_points(); ++q)
             {
-              const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
               switch (fluid_solid_reaction_scheme)
                 {
                   case no_reaction:
                   {
-                    Assert(out != nullptr,
```

Comment:
Not sure, but this code looks like it requires out to be non-null in all cases (even if the current version references the nullptr first). Maybe require out to be non-null and see if anything breaks?